scripts/gas-refund-program/computeMerkleTree.ts

```diff
@@ -35,22 +39,46 @@ export async function computeAndStoreMerkleTreeForChain({
       `merkle root for chainId=${chainId} epoch=${epoch} already exists`,
     );
 
-  const gasRefundParticipations = await GasRefundParticipation.findAll({
+  const gasRefundTXs = await GasRefundTransaction.findAll({
     where: { epoch, chainId },
+    /**
+     * in the future, guard against duplicate txs (of which multiple swaps can be a part of)
+     *
+     * ignore duplicates for cases like
+     * https://etherscan.io/tx/0xb10c51756678cb6cb1635ac339f9caa592f49ea6da936b109dbd49da8e0e0a6a
+     * where the graph returned three swaps for one tx, resulting
+     * in gas being fetched three times for one tx. only an issue while
+     * we get swaps not txs.
+     */
+    ...( epoch >= GasRefundDeduplicationStartEpoch ? {
+      attributes: [
+        [Sequelize.fn('DISTINCT', Sequelize.col('hash')), 'hash'], 'address', 'refundedAmountPSP'
+      ]
+    } : {})
   });
+  const addressRefunds: Record<string, BigNumber> = {}
+
+  for(let i = 0; i < gasRefundTXs.length; i++) {
+    const { address, refundedAmountPSP } = gasRefundTXs[i]
+    if (!addressRefunds[address]) {
+      addressRefunds[address] = new BigNumber(0)
+    }
+    addressRefunds[address] = addressRefunds[address].plus(new BigNumber(refundedAmountPSP))
+  }
+  const gasRefundParticipations: MinGasRefundParticipation[] = Object.entries(addressRefunds).map(([address, refundedAmountPSP]) => ({address, refundedAmountPSP: refundedAmountPSP.toFixed(0)}))
```

Comment:
You can replace all this treatment with a sql aggregation. It'll be more cpu/memory efficient + less error prone

Reply:
There are few examples here https://github.com/paraswap/paraswap-volume-tracker/pull/32/files#diff-873e1a96de8966552d6006f57abc47bcc7f8732d764c20f6f125d31d8eee7ca7R101